python/ray/data/tests/test_downstream_capacity_backpressure_policy.py

```diff
@@ -0,0 +1,138 @@
+import unittest
+from unittest.mock import MagicMock
+
+import pytest
+
+from ray.data._internal.execution.backpressure_policy.downstream_capacity_backpressure_policy import (
+    DownstreamCapacityBackpressurePolicy,
+)
+from ray.data._internal.execution.interfaces.physical_operator import PhysicalOperator
+from ray.data._internal.execution.operators.actor_pool_map_operator import (
+    ActorPoolMapOperator,
+)
+from ray.data._internal.execution.streaming_executor_state import OpState, Topology
+from ray.data.context import DataContext
+
+
+class TestDownstreamCapacityBackpressurePolicy(unittest.TestCase):
+    def setUp(self):
+        self.context = DataContext()
+        self.context.set_config("backpressure_task_slot_ratio", "2")
+        self.context.set_config("backpressure_max_queued_bundles", "4000")
+        self.mock_resource_manager = MagicMock()
+
+    def _mock_operator(
+        self,
+        op_class: PhysicalOperator,
+        num_enqueued_input_bundles=0,
+        num_active_tasks=0,
+    ):
+        """Helper method to create mock operator."""
+        mock_operator = MagicMock(spec=op_class)
+        mock_operator.num_active_tasks.return_value = num_active_tasks
+
+        op_state = MagicMock(spec=OpState)
+        op_state.total_enqueued_input_bundles.return_value = num_enqueued_input_bundles
+        return mock_operator, op_state
+
+    def _mock_actor_pool_operator(self, num_enqueued_input_bundles=0, num_task_slots=0):
+        """Helper method to create mock actor pool operator."""
+        mock_operator, mock_op_state = self._mock_operator(
+            ActorPoolMapOperator, num_enqueued_input_bundles, num_task_slots
+        )
+        mock_operator._actor_pool = MagicMock()
+        mock_operator._actor_pool.num_task_slots.return_value = num_task_slots
+        return mock_operator, mock_op_state
+
+    def _create_policy(
+        self, data_context: DataContext = None, topology: Topology = None
+    ):
+        """Helper method to create policy instance."""
+        context = data_context or self.context
+        return DownstreamCapacityBackpressurePolicy(
+            data_context=context,
+            topology=topology,
+            resource_manager=self.mock_resource_manager,
+        )
+
+    def test_can_add_input_no_backpressure_low_queue_size(self):
```

Comment:
Let's combine most of these tests into 1 parameterized by the queue length and num active tasks